components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -1511,44 +1491,26 @@ subroutine radiation_tend(state,   ptend,    pbuf,          cam_out, cam_in,  &
                      call get_curr_date(year,month,day,tod)
                      !populate with corresponding pbuf
                      !variables
-                     call pbuf_get_field(pbuf,aer_tau_bnd_sw_mon_1_idx, aer_tau_bnd_sw_mon_1)
-                     call pbuf_get_field(pbuf,aer_tau_bnd_sw_mon_2_idx, aer_tau_bnd_sw_mon_2)
-                     call pbuf_get_field(pbuf,aer_ssa_bnd_sw_mon_1_idx, aer_ssa_bnd_sw_mon_1)
-                     call pbuf_get_field(pbuf,aer_ssa_bnd_sw_mon_2_idx, aer_ssa_bnd_sw_mon_2)
-                     call pbuf_get_field(pbuf,aer_asm_bnd_sw_mon_1_idx, aer_asm_bnd_sw_mon_1)
-                     call pbuf_get_field(pbuf,aer_asm_bnd_sw_mon_2_idx, aer_asm_bnd_sw_mon_2)
-
+                     call pbuf_get_field(pbuf,aer_tau_bnd_sw_mon_idx, aer_tau_bnd_sw_mon)
+                     call pbuf_get_field(pbuf,aer_ssa_bnd_sw_mon_idx, aer_ssa_bnd_sw_mon)
+                     call pbuf_get_field(pbuf,aer_asm_bnd_sw_mon_idx, aer_asm_bnd_sw_mon)
+
+                     if (month==12) then
+                         next_month = 1
+                     else
+                         next_month = month + 1
+                     end if
                      if (current_month .ne. month) then
```

Comment:
I'm weirded out by the logic above and below this comment. If current_month and month are always set to the same thing, why not delete month and only use current_month? And is next_month getting set to the right value if current_month .ne. month?

Reply:
I think the idea here is that month is set by asking EAM to provide the actual month for this timestep.  current_month was meant to keep track of what month the last time the interpolation was done.  It made more sense when we were loading files on the fly, since you would only load new data if the "current_month" has changed, which would mean your currently loaded data was outdated.
As for `next_month`, if we set `current_month` before calculating `next_month` then it would be okay to use current_month in this calculation.  But since everything is being treated as dependent on `month` which is the only thing actually updated by timestep at this point, the code is using month here.  In one of my other comments I advocate for removing `current_month` unless it has some other use outside of the interpolation.

Reply:
Ok, that's what I thought. I agree we should remove current_month.